plugins/org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/typesystem/internal/RootResolvedTypes.java

```diff
@@ -167,6 +167,12 @@ protected boolean isIntentionalEarlyExit(/* @Nullable */ XExpression expression)
 	}
 	
 	protected AbstractDiagnostic createTypeDiagnostic(XExpression expression, LightweightTypeReference actualType, LightweightTypeReference expectedType) {
+		if (expectedType.isUnknown() && actualType.isAny())
+			return null;
+		
+		if (expectedType.isType(Object.class) && actualType.isUnknown())
```

Comment:
@Anton: If these case handling is neccessary here,  could you put a comment into the code, that explains the cases?